src/models/app_data_json_format.rs

```diff
@@ -163,24 +206,84 @@ mod tests {
         });
         let json: AppData = serde_json::from_value(value).unwrap();
         let expected = AppData {
-            version: "1.2.3".to_string(),
+            version: Version::V4,
+            app_code: "MooSwap".to_string(),
+            environment: Some("production".to_string()),
+            metadata: Some(Metadata {
+                referrer: Some(Referrer::V1(ReferrerV1 {
+                    address: "0x8c35B7eE520277D14af5F6098835A584C337311b"
+                        .parse()
+                        .unwrap(),
+                })),
+                quote: Some(Quote::V2(QuoteV2 { slippage_bips: 5 })),
+                ..Default::default()
+            }),
+        };
+
+        assert_eq!(json, expected);
+    }
+
+    #[test]
+    fn test_loading_json_v5() {
+        let value = json!({
+            "version": "0.5.0",
+            "appCode": "MooSwap",
+            "environment": "production",
+            "metadata": {
+                "referrer":{
+                    "version": "0.1.0",
+                    "address": "0x8c35B7eE520277D14af5F6098835A584C337311b",
+                },
+                "quote": {
+                    "version": "0.2.0",
+                    "slippageBips": "5",
+                },
+                "class": {
+                    "version": "0.1.0",
+                    "type": "limit",
+                },
```

Comment:
Is all this sub-versioning really necessary?

Reply:
This is what was specified in https://github.com/cowprotocol/cowswap/issues/1542

Reply:
ok... just seems like overkill. why not just version the outer scope?